pkgs/development/python-modules/dill/default.nix

```
@@ -1,48 +1,35 @@
 { lib
 , buildPythonPackage
 , fetchFromGitHub
-, pytestCheckHook
+, setuptools
 }:
 
 buildPythonPackage rec {
   pname = "dill";
-  version = "0.3.5.1";
+  version = "0.3.6";
+  format = "pyproject";
 
   src = fetchFromGitHub {
     owner = "uqfoundation";
     repo = pname;
     rev = "refs/tags/dill-${version}";
-    sha256 = "sha256-gWE7aQodblgHjUqGAzOJGgxJ4qx9wHo/DU4KRE6JMWo=";
+    hash = "sha256-lh1o/TqnqtYN9xTZom33y1/7ZhMEAFpheLdtalwgObQ=";
   };
 
-  checkInputs = [
-    pytestCheckHook
+  nativeBuildInputs = [
+    setuptools
   ];
 
-  # Tests seem to fail because of import pathing and referencing items/classes in modules.
-  # Seems to be a Nix/pathing related issue, not the codebase, so disabling failing tests.
-  disabledTestPaths = [
-    "tests/test_diff.py"
-    "tests/test_module.py"
-    "tests/test_objects.py"
-    "tests/test_session.py"
-  ];
-
-  disabledTests = [
-    "test_class_objects"
-    "test_importable"
-    "test_method_decorator"
-    "test_the_rest"
-    # test exception catching needs updating, can probably be removed with next update
-    "test_recursive_function"
-  ];
+  checkPhase = ''
+    python dill/tests/__main__.py
```

```suggestion
    ${python.interpreter} dill/tests/__main__.py
```

ALso, we should use the hooks.

```nix
  checkPhase = ''
    runHook preCheck
    ${python.interpreter} dill/tests/__main__.py
    runHook postCheck
  '';
```

---

I didn't know about `${python.interpreter}`. Thanks for teaching me about that and for catching the missing hooks.